coalib/core/Core.py

```diff
@@ -212,8 +212,8 @@ def _schedule_bears(self, bears):
         """
         for bear in bears:
             if self.dependency_tracker.get_dependencies(
-                    bear):  # pragma: no cover
-                logging.warning(
+                    bear):
+                raise RuntimeError(  # Unreachable code
```

Comment:
If we raise we need to break out immediately of the event loop. As the event loop will run on external processes, there may be other `_schedule_bears` functions running concurrently and trying to run the remaining bears.

I would suggest doing this:
```python
def _throw_exception(self, ex):
    # do cleanups, like unscheduling all bears by cleaning up self.running_tasks etc.

    # either raise here now:
    raise ex
    # or defer to self.run
    self._exception = ex

# inside self.run:
def run(self):
    try:
        self.event_loop.run_forever()
        if self._exception:
            raise self._exception
    finally:
        self.event_loop.close
```